### DISCUSSION TITLE
Upper limit for poll expires_in?

### DISCUSSION THREAD

**Discussion by obfuscatedgenerated** [mastodon/mastodon#20817] [Q&A] [answered]
Hello, 

I'm using the API to create a poll and I've discovered the bottom limit for expires_in as 300 seconds (5 min). I've also discovered there is an upper limit, but I'm not sure how long this is. I can't find this documented anywhere either.

Anyone know what the upper limit in seconds is for expires_in on creating polls via API.

Cheers 😊

**Comment by jeffsikes** [comment, marked as answer]
I was looking for this today, too! Found it.

5 minutes (300 seconds) is the minimum as you found out.
30 days (2,592,000 seconds) is the max.

Now we know! 

https://github.com/mastodon/mastodon/blob/2f7c3cb628ddf0438452325c52359f575aab6890/app/validators/poll_validator.rb#L4
